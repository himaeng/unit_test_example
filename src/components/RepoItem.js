import React, { PropTypes } from 'react';
import {
  Text,
  TouchableHighlight,
  View
} from 'react-native';

RepoItem.propTypes = {
  isSelected: PropTypes.bool,
  repo: PropTypes.object.isRequired,
  selectRepo: PropTypes.func
}

RepoItem.default = {
  isSelected: false
}

function RepoItem (props) {
  const { description, id, name, stargazers_count } = props.repo;
  const itemStyle = props.isSelected && [styles.item, styles.selected] || styles.item;
  return (
    <TouchableHighlight
      onPress={() => props.selectRepo(id)}
      underlayColor='#E0F2F1'
    >
      <View style={itemStyle}>
        <Text style={styles.title}>{name}</Text>
        <Text style={styles.stars}>{`${stargazers_count} stars`}</Text>
        { props.isSelected ? <Text>{description}</Text> : null }
      </View>
    </TouchableHighlight>
  )
}

const styles = {
  item: {
    paddingVertical: 8,
    paddingHorizontal: 24
  },
  selected: {
    backgroundColor: '#B2DFDB'
  },
  stars: {
    paddingBottom: 8
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    paddingBottom: 4
  }
}

export default RepoItem;
