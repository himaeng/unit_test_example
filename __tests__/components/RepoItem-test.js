import React from 'react';
import renderer from 'react-test-renderer';

import RepoItem from '../../src/components/RepoItem';
import { repos } from '../../config/jest/mockData';

describe('RepoItem', () => {
  it('ในกรณีที่ไม่คลิก หรือ isSelected = false', () => {
    const component = renderer.create(
      <RepoItem
        repo={repos[0]}
        selectRepo={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('ในกรณีที่คลิกเลือก หรือ isSelected = true', () => {
    const component = renderer.create(
      <RepoItem
        isSelected
        repo={repos[0]}
        selectRepo={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
