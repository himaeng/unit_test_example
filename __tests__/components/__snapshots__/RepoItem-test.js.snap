// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoItem ในกรณีที่คลิกเลือก หรือ isSelected = true 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#B2DFDB",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }
    >
      Test 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      50000 stars
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Useful description goes here.
    </Text>
  </View>
</View>
`;

exports[`RepoItem ในกรณีที่ไม่คลิก หรือ isSelected = false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "paddingHorizontal": 24,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 4,
        }
      }
    >
      Test 1
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 8,
        }
      }
    >
      50000 stars
    </Text>
  </View>
</View>
`;
